src/streamlink_cli/main.py

```diff
@@ -846,17 +849,28 @@ def log_current_arguments(session: Streamlink, parser: argparse.ArgumentParser):
             log.debug(f" {name}={value if name not in sensitive else '*' * 8}")
 
 
-def setup_logger_and_console(
-    stream=sys.stdout,
-    level: str = "info",
-    fmt: str | None = None,
-    datefmt: str | None = None,
-    file: str | None = None,
-    json=False,
-):
+def setup_console() -> None:
     global console
 
-    verbose = level in ("trace", "all")
+    console_output: TextIO | None
+    if args.quiet:
+        console_output = compat.devnull_txt
+    elif args.stdout or args.output == "-" or args.record == "-" or args.record_and_pipe:
+        # Console output should be on stderr if we are outputting a stream to stdout
```

Comment:
It's up to you but I don't think we need this comment. It's pretty clear to me what is happening here thanks to the rewritten code.